gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```
@@ -82,13 +82,33 @@ public Optional<ProxyBackendConfiguration> getBackendByName(String name)
     @Override
     public void deactivateBackend(String backendName)
     {
-        dao.deactivate(backendName);
+        updateBackendActivationStatus(backendName, false, () -> dao.deactivate(backendName));
     }
 
     @Override
     public void activateBackend(String backendName)
     {
-        dao.activate(backendName);
+        updateBackendActivationStatus(backendName, true, () -> dao.activate(backendName));
+    }
+
+    private void updateBackendActivationStatus(String clusterName, boolean newStatus, Runnable action)
+    {
+        GatewayBackend model = dao.findFirstByName(clusterName);
+        if (model == null) {
+            throw new IllegalStateException("No cluster found with name: " + clusterName + ", could not (de)activate");
+        }
+
+        boolean prevStatus = model.active();
+        action.run();
```

If the update fails for some reason, this will fail silently, making the next log line confusing. Should we only log when the update succeeds?

---

Do you mean the log line from `logActivationStatusChange` will be confusing? Since it only logs if the activation status changes though, doesn't that mean we would already only log if the activation status was correctly updated?